File: orion-viewer/src/universe/constellation/orion/viewer/device/EInkDevice.java
package universe.constellation.orion.viewer.device;

import android.view.View;

import universe.constellation.orion.viewer.OrionViewerActivity;
import universe.constellation.orion.viewer.prefs.GlobalOptions;

/**
 * Created by mike on 9/9/14.
 */
public abstract class EInkDevice extends AndroidDevice {

    protected int counter;

    @Override
    public void flushBitmap() {
        GlobalOptions options = ((OrionViewerActivity)activity).getGlobalOptions();
        if (options.isEinkOptimization()) {
            if (counter < options.getEinkRefreshAfter()) {
                doPartialUpdate(activity.getView());
                counter++;
            } else {
                counter = 0;
                doFullUpdate(activity.getView());

            }
        } else {
            super.flushBitmap();
        }
    }

    public void doPartialUpdate(View view) {
        super.flushBitmap();
    }

    public void doFullUpdate(View view) {
        super.flushBitmap();
    }
}

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/device/EInkDevice.kt
package universe.constellation.orion.viewer.device

import android.view.View

import universe.constellation.orion.viewer.OrionViewerActivity

abstract class EInkDevice : AndroidDevice() {

    private var counter: Int = 0

    private val einkOptimization: Boolean
        get() = options.isEinkOptimization

    override fun flushBitmap(view: View) {
        if (einkOptimization) {
            if (counter < options.einkRefreshAfter) {
                doPartialUpdate(view)
                counter++
            } else {
                counter = 0
                doFullUpdate(view)

            }
        } else {
            doDefaultUpdate(view)
        }
    }

    open fun doPartialUpdate(view: View) {
        super.flushBitmap(view)
    }

    open fun doFullUpdate(view: View) {
        super.flushBitmap(view)
    }

    open fun doDefaultUpdate(view: View) {
        super.flushBitmap(view)
    }

    @Throws(Exception::class)
    open fun doLighting(delta: Int): Int {
        return -1
    }
}
